web/controllers/notes_controller.ex

```diff
@@ -2,6 +2,13 @@ defmodule Notox.NoteController do
   use Notox.Web, :controller
   alias Notox.Note
 
+  def index(conn, %{"search" => %{"for" => keyword}}) do
```

Comment:
Thoughts on making this just `%{"query" => query}`? Seems a little simpler to me. If so, you'll have to figure out how to tell the form not to group the fields.